File: dashboard/src/types/cron.ts
/**
 * 定时任务状态常量
 */
export const CRON_JOB_STATUS = {
  ENABLED: 'enabled',
  DISABLED: 'disabled',
} as const

export type CronJobStatus = (typeof CRON_JOB_STATUS)[keyof typeof CRON_JOB_STATUS]

/**
 * 定时任务状态显示名称映射
 */
export const CronJobStatusLabels: Record<CronJobStatus, string> = {
  [CRON_JOB_STATUS.ENABLED]: '启用',
  [CRON_JOB_STATUS.DISABLED]: '禁用',
}

/**
 * 定时任务状态选项（用于下拉选择）
 */
export const CronJobStatusOptions = [
  { value: true, label: CronJobStatusLabels[CRON_JOB_STATUS.ENABLED] },
  { value: false, label: CronJobStatusLabels[CRON_JOB_STATUS.DISABLED] },
]

/**
 * 智能体简要信息（CronJob 关联）
 */
export interface AgentBrief {
  id: string
  name: string
}

/**
 * 定时任务配置
 */
export interface CronJob {
  id: string
  name: string
  cron_expression: string
  agent_id: string
  agent?: AgentBrief
  message: string
  enabled: boolean
  // 会话推送信息（通过智能体对话创建的定时任务才有）
  push_channel?: string // 推送渠道类型：qq/feishu
  push_bot_id?: string // 推送机器人ID
  push_user_id?: string // 推送目标用户ID
  push_group_id?: string // 推送目标群ID（群聊时）
  push_chat_id?: string // 会话ID（飞书 chat_id）
  created_at: string
  updated_at: string
}

/**
 * 创建定时任务请求
 */
export interface CreateCronJobRequest {
  name: string
  cron_expression: string
  agent_id: string
  message: string
  enabled?: boolean
}

/**
 * 更新定时任务请求
 */
export interface UpdateCronJobRequest {
  name: string
  cron_expression: string
  agent_id: string
  message: string
  enabled: boolean
}

File: beepbot/internal/types/cron.go
package types

import (
	"time"
)

// CronJobStatus 定时任务状态
type CronJobStatus bool

const (
	CronJobEnabled  CronJobStatus = true
	CronJobDisabled CronJobStatus = false
)

// CronJob 定时任务
type CronJob struct {
	ID             string        `json:"id" gorm:"column:id;primaryKey;type:varchar(64)"`
	Name           string        `json:"name" gorm:"column:name;type:varchar(128);not null;uniqueIndex"`
	CronExpression string        `json:"cron_expression" gorm:"column:cron_expression;type:varchar(64);not null"`
	AgentID        string        `json:"agent_id" gorm:"column:agent_id;type:varchar(64);index"`
	Message        string        `json:"message" gorm:"column:message;type:text"`
	Enabled        CronJobStatus `json:"enabled" gorm:"column:enabled;type:boolean;default:true;index"`

	// 会话推送信息（用于智能体处理完成后推送响应）
	// 只有用户通过智能体对话创建的定时任务才有推送信息
	PushChannel *string `json:"push_channel,omitempty" gorm:"column:push_channel;type:varchar(64)"`   // 推送渠道类型：qq/feishu
	PushBotID   *string `json:"push_bot_id,omitempty" gorm:"column:push_bot_id;type:varchar(64)"`     // 推送机器人ID
	PushUserID  *string `json:"push_user_id,omitempty" gorm:"column:push_user_id;type:varchar(64)"`   // 推送目标用户ID
	PushGroupID *string `json:"push_group_id,omitempty" gorm:"column:push_group_id;type:varchar(64)"` // 推送目标群ID（群聊时）
	PushChatID  *string `json:"push_chat_id,omitempty" gorm:"column:push_chat_id;type:varchar(64)"`   // 会话ID（飞书 chat_id）

	CreatedAt time.Time `json:"created_at" gorm:"column:created_at;type:timestamptz;not null"`
	UpdatedAt time.Time `json:"updated_at" gorm:"column:updated_at;type:timestamptz;not null"`
}

// CronQuery 定时任务查询参数
type CronQuery struct {
	Name    string // 名称模糊搜索
	Enabled *bool  // 是否启用（使用指针区分空值和 false）
}

// TableName 指定表名
func (CronJob) TableName() string {
	return "cron_jobs"
}
